### PR TITLE
v1.9 backports 2021-04-27

### PR DESCRIPTION
**Pull request by nebril**
* #14895 -- bugtool: Record attached BPF programs (@aditighag)
 * #15762 -- test: Skip K8sPolicy on GKE and 4.19 (@pchaigno)
 * #15838 -- contrib: Clean output of submit-backport script (@pchaigno)
 * #15841 -- .github: remove unnecessary docker hub credentials (@aanm)
 * #15836 -- docs/policy: Clarify table for deny policy scenarios (@pchaigno)
 * #15863 -- test/gke: refactor test-clusters operations (@nbusseneau)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 14895 15762 15838 15841 15836 15863 ; do contrib/backporting/set-labels.py $pr done 1.9; done
```

### CODE REVIEW

**Comment by nebril** [comment]
test-backport-1.9

**Review by aanm — APPROVED** [review]
Looks good for my change

**Review by aditighag — APPROVED** [review]
>#14895 -- bugtool: Record attached BPF programs (@aditighag)

lgtm

**Comment by nebril** [comment]
all gke builds are failing rn, will rerun when the fix is out

**Comment by nebril** [comment]
test-gke

**Review by pchaigno — APPROVED** [review]
My changes look good :+1:

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-backport-1.9

**Review by gandro — APPROVED** [review]

File: Documentation/contributing/testing/e2e.rst [hunk 1]
```diff
@@ -503,9 +503,9 @@ cluster.
 2- Invoke the tests from ``cilium/test`` with options set as explained in
 `Running End-To-End Tests In Other Environments via kubeconfig`_
 
-.. note:: GKE clusters may have namespaces stuck at the ``Terminating`` state,
-          causing Ginkgo tests to fail. If so, you'll see them in ``kubectl get ns``,
-          and can get rid of them by running ``cilium/test/gke/clean-cluster.sh``.
+.. note:: The tests require the ``NATIVE_CIDR`` environment variable to be set to
+          the value of the cluster IPv4 CIDR returned by the ``gcloud container
+          clusters describe`` command.
```
Comment (nbusseneau) [inline Documentation/contributing/testing/e2e.rst]: The original commit only deleted the part on the left, not sure why this bit got added in.

Comment (nebril) [reply Documentation/contributing/testing/e2e.rst]: Will delete, I guess I messed up the conflict resolution.

**Comment by nebril** [comment]
test-backport-1.9

**Comment by nebril** [comment]
test-backport-1.9

**Comment by nebril** [comment]
test-backport-1.9

**Comment by joestringer** [comment]
Hit what looks to be a pretty rare flake on GKE (#13752): https://jenkins.cilium.io/job/Cilium-PR-K8s-GKE/5348/ . Otherwise looks fine. Merging.

**Pull request merged**